code-challenges.test.js

```diff
@@ -26,42 +26,162 @@ const numbersArray2 = [24, 27, 30, 33, 36]
+
+// // output: PASS  ./code-challenges.test.js
+//   threeDivider
+//   ✓  takes a object as an argument and decides if the number inside it is evenly divisible by three or not (3 ms)
+// upperCaser
+//   ✓ takes in an array of words and returns an array with all the words capitalized (1 ms)
+
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
-// Pseudo code:
+// Pseudo code: 
+    // Input: a string
+    // Output: the index number of the first vowel in the input string
+    // Process: create a function that uses a for loop to iterate through a string and checks whether the string contains any vowels from the variable vowels = ["a", "e", "i", "o", "u"]
 
 // a) Create a test with expect statements for each of the variables provided.
 
+describe("firstVowel", () => {
+    it("takes in a string and logs the index of the first vowel", () => {
 const vowelTester1 = "learn"
 // Expected output: 1
 const vowelTester2 = "academy"
 // Expected output: 0
 const vowelTester3 = "challenges"
 // Expected output: 2
+        expect(firstVowel(vowelTester1)).toEqual(1)
+        expect(firstVowel(vowelTester2)).toEqual(0)
+        expect(firstVowel(vowelTester3)).toEqual(2)
+    })
+})
+
+// test output: FAIL  ./code-challenges.test.js
+//   threeDivider
+//   ✓  takes a object as an argument and decides if the number inside it is evenly divisible by three or not (2 ms)
+// upperCaser
+//   ✓ takes in an array of words and returns an array with all the words capitalized (1 ms)
+// firstVowel
+//   ✕ takes in a string and logs the index of the first vowel (1 ms)
+
+// ● firstVowel › takes in a string and logs the index of the first vowel
+
+//   ReferenceError: firstVowel is not defined
 
 // b) Create the function that makes the test pass.
+
+const vowels = ["a", "e", "i", "o", "u"]
+
+const firstVowel = (string) => {
+   for (let i = 0; i < string.length; i++) {
+        if (vowels.includes(string[i])) {
+            return i;
+        }
+   }}
+//  return vowels.includes(string[i]) ? i : "There is no vowel in this string" 
+
+// output:  PASS  ./code-challenges.test.js
+//   threeDivider
+//   ✓  takes a object as an argument and decides if the number inside it is evenly divisible by three or not (2 ms)
+// upperCaser
+//   ✓ takes in an array of words and returns an array with all the words capitalized
+// firstVowel
+//   ✓ takes in a string and logs the index of the first vowel
+
+// notes: I tried and tried to figure out a way to use a ternary operator for this function, but I cannot find a way to make it work. After doing a bit of research, it seems it wont work because the return is still inside the for loop, so I tried making the ternary into a variable, and calling it after the for loop, but still with no success. It possibly has something to do with the 'else' part of the ternary being attached?
```

Comment:
That's partially correct, the ternary won't work if you're just trying to return the results of the ternary within the for loop, because it will only ever check the first letter then end the iteration. However, if you really wanted to use a ternary, there are ways to incorporate a ternary. I will provide one approach at the bottom of this feedback block, so if you don't want to see an answer yet, don't look at the bottom!
z
z
z
z
z
z
z
z
z
z
z
z
z
z
z
z
z
`const firstVowel = (str) => {
    let newArr = []
    for(let i = 0; i < str.length; i++) {
        'aeiou'.includes(str[i]) ? newArr.push(i) : i
    }
    return Math.min(...newArr)
}`